windows/src/global/inc/keymanengine.h

```diff
@@ -0,0 +1,254 @@
+PWSTR strtowstr(PSTR in);
+
+WCHAR MapVirtualKeys(WORD keyCode, UINT shiftFlags);
+
+void SelectApplicationIntegration();   // I4287
+
+void PostDummyKeyEvent();  // I3301 - Handle I3250 regression with inadvertent menu activation with Alt keys   // I3534   // I4844
+
+/* Debugging functions */
+
+#ifndef SendDebugMessage
+   // I4379
+typedef enum ATSDMState { sdmInternat, sdmAIDefault, sdmMessage, sdmKeyboard, sdmGlobal, sdmMenu, sdmDebug, sdmLoad, sdmOther } TSDMState;
+
+void InitDebugging();
+void UninitDebugging();
+
+extern "C" void _declspec(dllexport) WINAPI Keyman_WriteDebugEvent(char* file, int line, PWCHAR msg);
+
+#define SendDebugMessage(hwnd,state,kmn_lineno,msg) (ShouldDebug((state)) ? SendDebugMessage_1((hwnd),(state),(kmn_lineno), __FILE__, __LINE__, (msg)) : 0)
+#define SendDebugMessageFormat(hwnd,state,kmn_lineno,msg,...) (ShouldDebug((state)) ? SendDebugMessageFormat_1((hwnd),(state),(kmn_lineno), __FILE__, __LINE__, (msg),__VA_ARGS__) : 0)
+#define ShouldDebug(state) ShouldDebug_1()
+#define DebugLastError(context) (DebugLastError_1(GetLastError(), (context), __FILE__,__LINE__,__FUNCTION__))
+#define DebugLastError0(error, context) (DebugLastError_1((error), (context), __FILE__,__LINE__,__FUNCTION__))
+int SendDebugMessage_1(HWND hwnd, TSDMState state, int kmn_lineno, char* file, int line, char* msg);
+int SendDebugMessageFormat_1(HWND hwnd, TSDMState state, int kmn_lineno, char* file, int line, char* fmt, ...);
+void DebugLastError_1(DWORD err, char* context, char* file, int line, char* func);
+void DebugMessage(LPMSG msg, WPARAM wParam);
+void DebugShift(char* function, char* point);
+BOOL DebugSignalPause(BOOL fIsUp);
+char* Debug_VirtualKey(WORD vk);
+char* Debug_UnicodeString(PWSTR s, int x = 0);
+
+BOOL ShouldDebug_1(); // TSDMState state);
+
+#endif
+
+#ifdef _DEBUG
+#define OutputThreadDebugString(s) _OutputThreadDebugString(s)
+void _OutputThreadDebugString(char* s);
+#else
+#define OutputThreadDebugString(s)
+#endif
+
+/* Keyboard selection functions */
+
+void HandleRefresh(int code, LONG tag);
+void RefreshKeyboards(BOOL Initialising);
+void CheckScheduledRefresh();
+void ScheduleRefresh();
+void ReleaseKeyboards(BOOL Lock);
+void CheckScheduledRefresh();
+void ScheduleRefresh();
+
+/* Glossary conversion functions */
+
+HKL ForceKeymanIDToHKL(DWORD KeymanID); // sign extends DWORD to HKL on Win64
+DWORD ForceHKLToKeymanID(HKL hkl);      // truncates HKL to DWORD on Win64
+   // I4220
+DWORD HKLToKeyboardID(HKL hkl);
+WORD HKLToLanguageID(HKL hkl);
+WORD HKLToLayoutNumber(HKL hkl);
+WORD HKLToLayoutID(HKL hkl);
+DWORD EthnologueCodeToKeymanID(DWORD EthCode);
+DWORD EthnologueStringCodeToDWord(PWSTR EthCode);
+
+PWSTR  GetSystemStore(LPKEYBOARD kb, DWORD SystemID);
+
+DWORD ExceptionMessage(LPSTR Proc, LPEXCEPTION_POINTERS ep);
+
+void keybd_shift(LPINPUT pInputs, int* n, BOOL isReset, LPBYTE const kbd);
+
+//#define KEYEVENT_EXTRAINFO_KEYMAN 0xF00F0000   // I4370
+#ifndef _KEYMAN64_LIGHT
```

Comment:
Should we update this define (and therefore all the places it is defined in projects) to KEYMANENGINE_LIGHT?

Reply:
In theory we should be able to remove it once we finish all the refactoring as I reckon its use is another example of this patching.